We.Sell.Bread.API.Integration.Tests/Tests/CustomerControllerTests.cs

```diff
@@ -1,78 +1,211 @@
+            responseContent.Should().Be("One or more customer details were invalid");
         }
 
-       
+        [Fact]
+        public async Task GivenEmptyNameWhenCreatingCustomerReturnInternalServerErrorAsync()
+        {
+            var updateCommand = CustomerCommand();
+
+            var response = await _httpClient.PostAsJsonAsync($"/Customer/Post", updateCommand);
+
+            response.StatusCode.Should().Be(HttpStatusCode.InternalServerError);
+        }
+
+        [Fact]
+        public async Task GivenDataExistWhenRetrievingAllCustomerReturnOkStatusCodeAsync()
+        {
+            var response = await _httpClient.GetAsync("/Customer/GetAll");
+
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public async Task CRUDTestAsync()
+        {
+            //create
+            var customerName = "Zanele Khumalo";
+            var createCommand = CustomerCommand(customerName);
+
+            var response = await _httpClient.PostAsJsonAsync($"/Customer/Post", createCommand);
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            var responseContent = await response.Content.ReadAsStringAsync();
+            var customer = JsonUtils.DeserializeToDto<CustomerDto>(responseContent);
+            customer.CustomerName.Should().Be(customerName);
+
+            //read
+            var getResponse = await _httpClient.GetAsync($"/Customer/Get/{customer.Id}");
+
+            getResponse.IsSuccessStatusCode.Should().BeTrue();
+            getResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            var getResponseContent = await getResponse.Content.ReadAsStringAsync();
+            var getCustomer = JsonUtils.DeserializeToDto<CustomerDto>(getResponseContent);
+            getCustomer.CustomerName.Should().Be(customerName);
+
+            //update
+            var updatedCustomerName = "Zanele Ayanda Khumalo";
+            var updateCommand = CustomerCommand(updatedCustomerName);
+
+            var updateResponse = await _httpClient.PutAsJsonAsync($"/Customer/Put/{customer.Id}", updateCommand);
+            updateResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            var updateResponseContent = await updateResponse.Content.ReadAsStringAsync();
+            var updateCustomer = JsonUtils.DeserializeToDto<CustomerDto>(updateResponseContent);
+            updateCustomer.CustomerName.Should().Be(updatedCustomerName);
+
+            //delete
+            var deleteResponse = await _httpClient.DeleteAsync($"/Customer/Delete/{customer.Id}");
+
+            deleteResponse.IsSuccessStatusCode.Should().BeTrue();
+            deleteResponse.StatusCode.Should().Be(HttpStatusCode.NoContent);
+
+        }
+
+        private static CustomerCommand CustomerCommand(string? customerName = null)
+        {
+            var contactNo = "0456871245";
+            var emailAddress = "foo@far.com";
```

Comment:
Because we do have an existing library for fake data generation, let keep the consistency and use the library

Reply:
Fixed